tests/selection-playlist.spec.ts

```diff
@@ -0,0 +1,273 @@
+    await pages.advancedConnectionPage.goToMainMenu();
+  });
+
+  await test.step('Go to the Song languages - choose correct one', async () => {
+    await pages.mainMenuPage.goToManageSongs();
+    await pages.manageSongsPage.goToSelectSongLanguage();
+    await pages.songLanguagesPage.ensureSongLanguageIsSelected(engLanguage);
+    await pages.songLanguagesPage.goBackToMainMenu();
+  });
+
+  await test.step('Go to the Song list and choose the song - ensure song is not visible in Selection playlist yet', async () => {
+    await pages.mainMenuPage.goToSingSong();
+    await pages.songListPage.expectPlaylistToBeSelected(selectionPlaylist);
+    await expect(pages.songListPage.getSongElement(unknownSong.ID)).not.toBeVisible();
+    await pages.songListPage.goToPlaylist(engPlaylist);
+    await pages.songListPage.openPreviewForSong(unknownSong.ID);
+  });
+
+  await test.step('Play the song and after above 80% complete - exit the song', async () => {
+    await pages.songPreviewPage.goNext();
+    await pages.songPreviewPage.playTheSong();
+    await pages.gamePage.skipIntroIfPossible();
+    await page.waitForTimeout(7000);
+    await pages.gamePage.exitSong();
+  });
+
+  await test.step('Skip to the Song list', async () => {
+    await pages.postGameResultsPage.skipScoresAnimation();
+    await pages.postGameResultsPage.goToHighScoresStep();
+    await pages.postGameHighScoresPage.goToSongList();
+  });
+
+  await test.step('A song that is more than 80% complete, should be added to the Selection playlist', async () => {
+    await pages.songListPage.goToPlaylist(selectionPlaylist);
+    await expect(pages.songListPage.getSongElement(unknownSong.ID)).toBeVisible();
+  });
+});
+
+test('A song that is less than 80% complete is not adding to the Selection playlist', async ({ page }) => {
+  await page.goto('/?e2e-test');
+  await pages.landingPage.enterTheGame();
+
+  await test.step('Select Advanced setup', async () => {
+    await pages.inputSelectionPage.selectAdvancedSetup();
+    await pages.advancedConnectionPage.goToMainMenu();
+  });
+
+  await test.step('Ensure song language is selected', async () => {
+    await pages.mainMenuPage.goToSingSong();
+    await pages.songLanguagesPage.ensureSongLanguageIsSelected(engLanguage);
+    await pages.songLanguagesPage.continueAndGoToSongList();
+  });
+
+  await test.step('Search the song - expect song is not visible in Selection playlist yet', async () => {
+    await pages.songListPage.expectPlaylistToBeSelected(selectionPlaylist);
+    await pages.songListPage.closeTheSelectionPlaylistTip();
+    await expect(pages.songListPage.getSongElement(unknownSong.ID)).not.toBeVisible();
+    await pages.songListPage.searchSong(unknownSong.title);
+  });
+
+  await test.step('Toggle game mode to `Cooperation` and play the song', async () => {
+    await pages.songListPage.openPreviewForSong(unknownSong.ID);
+    await pages.songPreviewPage.toggleGameMode();
```

Comment:
is this needed for the test?

Reply:
In this test, I am waiting for some amount of points and I have already prepared a method for it (using cooperation score)